apache/trafficcontrol
# TP: by default only show parameter diff when comparing 2 profiles

## What does this PR (Pull Request) do?

This PR changes the default behavior when comparing the parameters of 2 profiles:

current behavior (#3919 ): ALL parameters (shared and not shared) are displayed.

new behavior: As requested by users, ONLY the parameters that belong to one profile and NOT the other (the diff) are displayed by DEFAULT and the user is given the option to "Show Shared Params" if so desired.


## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?

- Launch TP

1. 
- Navigate to https://tp.domain.com/#!/profiles
- Select "Compare Profiles" under the More button
- Select 2 profiles to compare
- Ensure none of the "shared" parameters are displayed
- add or remove parameters from each profile; click "Update"
- ensure the comparison is updated correctly after confirmation.

2.
- Navigate to https://tp.domain.com/#!/profiles
- Select "Compare Profiles" under the More button
- Select 2 profiles to compare
- Select 'Show Shared Params'
- Ensure each profile has the proper parameters "checked". (you can double check by checking the parameters for the profile at https://tp.domain.com/#!/profiles/{profile_id}/parameters)
- add or remove parameters from each profile; click "Update"
- ensure the proper parameters are assigned to each profile after confirmation.

## The following criteria are ALL met by this PR

- [x] The tests associated with #3919 cover this new feature however a minor change was required
- [x] The CHANGELOG.md entry for #3919 is still applicable
